toolchain/lc-compile/src/grammar.g

```diff
@@ -854,6 +859,34 @@
             Statements(-> Body)*/
 
 --------------------------------------------------------------------------------
+-- Bytecode Syntax
+--------------------------------------------------------------------------------
+
+'nonterm' Bytecodes(-> BYTECODE)
+
+    'rule' Bytecodes(-> sequence(Left, Right)):
+        Bytecode(-> Left) Separator
+        Bytecodes(-> Right)
+
+    'rule' Bytecodes(-> nil):
+        -- empty
+
+'nonterm' Bytecode(-> BYTECODE)
+
+    'rule' Bytecode(-> label(Position, Name)):
+        Identifier(-> Name) @(-> Position) ":"
+
+    'rule' Bytecode(-> variable(Position, Name, Type)):
+        "variable" @(-> Position) Identifier(-> Name) OptionalTypeClause(-> Type)
```

Comment:
My original thinking here was that since variables are the same inside a bytecode block as outside - they are named, have a type, have block-scope - I figured it was suitable. However, there is a difference in that variables are not automatically 'reset' in bytecode context. This is probably enough justification that they should be described differently.